engine/crates/engine-v2/schema/src/builder/input_values.rs

```diff
@@ -1,48 +1,48 @@
 use federated_graph::Value;
 
-use crate::{SchemaInputValue, SchemaInputValues};
+use crate::{SchemaInputValueRecord, SchemaInputValues};
 
 use super::BuildContext;
 
 #[derive(Debug, Clone, Copy)]
 pub(crate) struct InaccessibleEnumValue;
 
 impl SchemaInputValues {
-    pub(crate) fn ingest_arbitrary_federated_value(
+    pub(crate) fn ingest_as_json(
         &mut self,
         ctx: &BuildContext,
         value: Value,
-    ) -> Result<SchemaInputValue, InaccessibleEnumValue> {
+    ) -> Result<SchemaInputValueRecord, InaccessibleEnumValue> {
         Ok(match value {
-            Value::Null => SchemaInputValue::Null,
-            Value::String(id) | Value::UnboundEnumValue(id) => SchemaInputValue::String(id.into()),
-            Value::Int(n) => SchemaInputValue::BigInt(n),
-            Value::Float(f) => SchemaInputValue::Float(f),
-            Value::Boolean(b) => SchemaInputValue::Boolean(b),
+            Value::Null => SchemaInputValueRecord::Null,
+            Value::String(id) | Value::UnboundEnumValue(id) => SchemaInputValueRecord::String(id.into()),
```

Comment:
Not really a comment on this PR really, but converting enum values into strings is often wrong?

I assume it's fine here because we put these into variables and not the query literal?  But might be worth a comment saying that if so.

Reply:
But then if we try and write this SchemaInputValueRecord back into GraphQL it'll be `{a: "ACTIVE"}` which is not the same?  Is there some way in which that is not a bug?

Reply:
hum, that's a good remark. Today it's not a problem since we always pass everything through variables, so serialized as JSON. Will do it in another PR then, you're right that I don't see how we could no keep that case...

Reply:
> Today it's not a problem since we always pass everything through variables

Yeah, this is what I was saying in my initial comment - I assume it's not actually a problem, but a comment here explaining why its not a problem would clarify (although if you want to properly fix it then that is also cool)

Reply:
yeah good point, taking a look on it, it's probably best to just keep track of them and only treat them as strings when serializing